sqlglot/expressions.py

```diff
@@ -7036,11 +7039,15 @@ def _apply_cte_builder(
     append: bool = True,
     dialect: DialectType = None,
     copy: bool = True,
+    scalar: bool = False,
     **opts,
 ) -> E:
     alias_expression = maybe_parse(alias, dialect=dialect, into=TableAlias, **opts)
     as_expression = maybe_parse(as_, dialect=dialect, **opts)
-    cte = CTE(this=as_expression, alias=alias_expression, materialized=materialized)
+    if scalar and not isinstance(as_expression, Subquery):
+        # scalar CTE must be wrapped in a subquery
+        as_expression = Subquery(this=as_expression)
```

Comment:
maybe_copy(as_expression, copy=copy),

Reply:
we don't want to mutate as_expression (this sets the parent)

Reply:
but `as_expression` is parsed earlier, i.e. it's probably a "copy" already?

Reply:
ah, ok, it `maybe_parse`d

Reply:
you're right, let's do that instead, pass copy into maybe_parse